handler/auth.go

```diff
@@ -141,8 +130,11 @@ func (h *Handler) Authenticate(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var user model.User
-	h.db.First(&user, "username = ?", request.Username)
+	user, err := h.usersRepo.GetByUsername(request.Username)
+	if err != nil {
+		// Something went wrong
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Will this 500 on a not found?

Reply:
yes, it would. I'll check for users.ErrNotFound before 500ing